mongo_connector/oplog_manager.py

```diff
@@ -106,8 +106,25 @@ def fields(self, value):
             self._fields = set(value)
             # Always include _id field
             self._fields.add('_id')
+
```

Comment:
It is possible that the user has only set fields using the constructor, and not this attribute. Anything you do here must also happen in the constructor (line 87) since it's possible that this will never get called, and self._fieldsdict will cause an AttributeError in pop_excluded_fields (line 369)